servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HttpResponseDecoder.java

```diff
@@ -41,17 +43,19 @@
 import static io.servicetalk.http.api.HttpResponseStatus.SWITCHING_PROTOCOLS;
 import static io.servicetalk.http.api.HttpResponseStatus.StatusClass.INFORMATIONAL_1XX;
 import static io.servicetalk.transport.netty.internal.CloseHandler.UNSUPPORTED_PROTOCOL_CLOSE_HANDLER;
-import static java.lang.Character.isISOControl;
 import static java.lang.Math.min;
 import static java.nio.charset.StandardCharsets.US_ASCII;
 import static java.util.Objects.requireNonNull;
 
 final class HttpResponseDecoder extends HttpObjectDecoder<HttpResponseMetaData> {
 
-    private static final byte[] FIRST_BYTES = "HTTP".getBytes(US_ASCII);
-    private static final ByteProcessor ENSURE_NO_CONTROL_CHARS = value -> {
-        if (value != HT && isISOControl(value)) {   // allow HTAB
-            throw newIllegalCharacter(value);
+    private static final byte[] FIRST_BYTES = "HTTP/".getBytes(US_ASCII);
+    private static final byte DEL = 127;
+
+    private static final ByteProcessor ENSURE_REASON_PHRASE = value -> {
+        // Any control character (0x00-0x1F) except HT
+        if (((value & 0xE0) == 0 && value != HT) || value == DEL) {
+            throw new IllegalCharacterException(value, "HTAB / SP / VCHAR / obs-text");
```

Comment:
since we will wrap this immediately with a `DecoderException` can we use `ThrowableUtils.unknownStackTrace()` here?